documentation/src/docs/wire-protocol.md

```diff
@@ -20,54 +20,87 @@ All the requests and replies have 8 byte headers with two fields (all data is wr
 - **Length**:  An Integer (4 bytes) (Messages should be less than 2<sup>24</sup>, but the upper bits remain zero). Payload size of the message (possibly zero, indicating there is no data). The remainder of the fields are specific to the type of the message. A few important messages are listed below.
 
+## Protocol Primitive Types
+
+The protocol is built out of the following primitive types.
+
+| **Type** | **Description** |
+|----------|--------------|
+|BOOLEAN (1 byte)| Values 0 and 1 are used to represent _False_ and _True_ respectively. When reading a boolean value, any non-zero value is considered true.|
+|STRING (2 bytes)|A sequence of characters. The first 2 bytes are used to indicate the byte length of the UTF-8 encoded character sequence, which is non-negative. This is followed by the UTF-8 encoding of the string.|
+|VARLONG (8 bytes)|An Integer between -2<sup>63</sup> and 2<sup>63</sup>-1 inclusive. Encoding follows the variable-length zig-zag encoding from [Google Protocol Buffers](https://developers.google.com/protocol-buffers/).|
+|INT (4 bytes)|An Integer between -2<sup>31</sup> and 2<sup>31</sup>-1 inclusive.|
+|UUID (16 bytes)|Universally Unique Identifiers (UUID) as defined by RFC 4122, ISO/IEC 9834-8:2005, and related standards. It can be used as a global unique 128-bit identifier.|
+
+
 # Reading
 
 ## Read Segment - Request
 
-1. `Segment`: String (2 bytes) followed by that many bytes of Java's Modified UTF-8. This Segment indicates the Stream Segment that was read.
-2. `Offset`: Long (8 bytes). The `Offset` in the Stream Segment to read from.
-3. `suggestedLength` of Reply: Integer (4 bytes). The clients can request for the required length to the server (but the server may allot a different number of bytes.
-4. `delegationToken`: String (2 byte) followed by that many bytes of Java's Modified UTF-8. This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.
+| **Field**   |**Datatype**   | **Description**     |
+|-------------|------------|----------|
+|  `Segment`| String| The Stream Segment that was read. |
+| `Offset`   | Long| The `Offset` in the Stream Segment to read from. |
+| `suggestedLength` of Reply|Integer|The clients can request for the required length to the server (but the server may allot a different number of bytes.|
+|`delegationToken`|String| This was added to perform _auth_. It is an opaque-to-the-client token provided by the Controller that says it's allowed to make this call.|
+|`RequestId`| Long| The client-generated _ID_ that identifies a client request.|
+
+More information on `ReadSegment` Request messages like `MergeSegment`, `SealSegment`, `TruncateSegment` and `DeleteSegment`, can be found [here](https://github.com/pravega/pravega/blob/master/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java).
```

Comment:
We should change it to something of the pattern "More information on `Segment` Request messages..."

Reply:
Edited